src/goose/specs.clj

```diff
@@ -106,6 +106,11 @@
                    ::cron-schedule]
           :opt-un [::timezone]))
 
+;;; ============== Batch Opts ==============
+(s/def ::callback-fn-sym (s/nilable ::fn-sym))
```

Comment:
Why do we need multi arity function for callback?

Reply:
I was thinking we'll give callbacks 2 info: first arg will be a `batch-id` & second arg will be status on completion (success/dead/partial)
If we're passing in a map instead, user will have to know the keys for destructuring the map. So, I am suggesting 2-arity callback-fn

Reply:
We. can add these details when we are implementing the callback in the flow? Depending on how its called, with what args, this spec can be updated?
wdyt?

Reply:
But yeah, the args you highlighted make sense.
The thing is, its goose code which is calling the callback fn. That cannot be altered by the user. So what would be the advantage of giving 2-arity fn input? Whatever call is being made in the code is the one that needs to be supported

Reply:
@olttwa Passing the batch's status to the callback fn would mean that it is responsible to handle those cases as well? 

Shall we defer making changes to the `callback-fn-sym` spec until we implement callbacks? This might be a longer discussion.

Reply:
Moving this conversation to Slack.